ios/Sources/DownloadPlugin.swift

```diff
@@ -0,0 +1,76 @@
+import Combine
+import DownloadManagerKit
+import SwiftRs
+import Tauri
+import WebKit
+
+class CreateArgs: Decodable {
+   let key: String
+   let url: String
+   let path: String
+}
+
+class KeyArgs: Decodable {
+   let key: String
+}
+
+class DownloadPlugin: Plugin {
+   let downloadManager = DownloadManager.shared
+   
+   override init()
+   {
+      super.init()
+      Task {
+          for await download in DownloadManager.shared.changed {
+             try? self.trigger("changed", data: download);
+             Logger.debug("[\(download.key)] \(download.state) - \(String(format: "%.0f", download.progress))%")
```

Comment:
Do we need this logging statement? If there are multiple downloads happening, depending on frequently these changes are triggered, it seems like this could be a bottleneck/unnecessary slowdown, and also lead to a ton of logging.

Reply:
Good question. I've now surrounded this with conditional directives so it is only used when debugging.